### PR TITLE
Add table oci_autoscaling_auto_scaling_policy

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  p.display_name,
  p.id,
  p.auto_scaling_configuration_id,
  c.cool_down_in_seconds,
  c.max_resource_count,
  c.min_resource_count
from
  oci_autoscaling_auto_scaling_policy as p,
  oci_autoscaling_auto_scaling_configuration as c
where
  p.auto_scaling_configuration_id = c.id
+----------------------------------+--------------------------------------+-------------------------------------------------------------------------------------------------------------+------------------>
| display_name                     | id                                   | auto_scaling_configuration_id                                                                               | cool_down_in_seco>
+----------------------------------+--------------------------------------+-------------------------------------------------------------------------------------------------------------+------------------>
| autoscaling-policy-20230602-1245 | d5da759e-1272-4637-bd5a-e17afa10343c | ocid1.autoscalingconfiguration.oc1.ap-mumbai-1.aaaaaaaawedcvphv7yosjoegjbxph2wnsewkntwrm657gacdeyclnhddcj2q | 300              >
+----------------------------------+--------------------------------------+-------------------------------------------------------------------------------------------------------------+------------------>

Time: 3.2s. Rows fetched: 2. Hydrate calls: 1.
```
</details>

### CODE REVIEW

**Comment by ParthaI** [comment]
Reference: https://github.com/turbot/steampipe-plugin-oci/pull/445

File: oci/table_oci_autoscaling_auto_scaling_policy.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+	"github.com/oracle/oci-go-sdk/v65/autoscaling"
+	"github.com/oracle/oci-go-sdk/v65/common"
+	"github.com/turbot/go-kit/types"
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+// // TABLE DEFINITION
```
Comment (madhushreeray30) [inline oci/table_oci_autoscaling_auto_scaling_policy.go:14]:
```suggestion
//// TABLE DEFINITION
```

File: oci/table_oci_autoscaling_auto_scaling_policy.go [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+					Name:    "display_name",
+					Require: plugin.Optional,
+				},
+			},
+		},
+		GetMatrixItemFunc: BuildCompartementRegionList,
+		Columns: []*plugin.Column{
+			{
+				Name:        "display_name",
+				Description: "A user-friendly name. Does not have to be unique, and it's changeable",
```
Comment (madhushreeray30) [inline oci/table_oci_autoscaling_auto_scaling_policy.go:42]:
```suggestion
				Description: "A user-friendly name. Does not have to be unique, and it's changeable.",
```

File: oci/table_oci_autoscaling_auto_scaling_policy.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+				Type:        proto.ColumnType_BOOL,
+			},
+			{
+				Name:        "policy_type",
+				Description: "The type of autoscaling policy.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "time_created",
+				Description: "Time that Auto Scaling Policy was created.",
```
Comment (madhushreeray30) [inline oci/table_oci_autoscaling_auto_scaling_policy.go:67]:
```suggestion
				Description: "Time that autoscaling policy was created.",
```

File: oci/table_oci_autoscaling_auto_scaling_policy.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+	Id                         *string
+	DisplayName                *string
+	TimeCreated                *common.SDKTime
+	Capacity                   *autoscaling.Capacity
+	IsEnabled                  *bool
+	PolicyType                 *string
+	AutoScalingConfigurationId *string
+}
+
+// // LIST FUNCTION
```
Comment (madhushreeray30) [inline oci/table_oci_autoscaling_auto_scaling_policy.go:108]:
```suggestion
//// LIST FUNCTION
```

File: oci/table_oci_autoscaling_auto_scaling_policy.go [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+	// handle empty id in get call
+	if id == "" || configurationId == "" {
+		return nil, nil
+	}
+
+	// Create Session
+
+	session, err := autoScalingService(ctx, d, region)
+	if err != nil {
+		logger.Error("getAutoscalingAutoScalingPolicy", "error_AutoscalingService", err)
```
Comment (madhushreeray30) [inline oci/table_oci_autoscaling_auto_scaling_policy.go:194]:
```suggestion
		logger.Error("oci_autoscaling_auto_scaling_policy.getAutoscalingAutoScalingPolicy", "connection_error", err)
```

File: oci/table_oci_autoscaling_auto_scaling_policy.go [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+	}
+
+	return nil, err
+}
+
+// // HYDRATE FUNCTION
+func getAutoscalingAutoScalingPolicy(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	region := d.EqualsQualString(matrixKeyRegion)
+	logger.Debug("getAutoscalingAutoScalingPolicy", "OCI_REGION", region)
```
Comment (madhushreeray30) [inline oci/table_oci_autoscaling_auto_scaling_policy.go:173]:
```suggestion
	logger.Debug("oci_autoscaling_auto_scaling_policy.getAutoscalingAutoScalingPolicy", "OCI_REGION", region)
```

**Review by madhushreeray30 — COMMENTED** [review]
@ParthaI can you please have a look at the comments, thanks!

File: oci/table_oci_autoscaling_auto_scaling_policy.go [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAutoscalingAutoScalingPolicy,
+			},
+
+			// Standard Steampipe columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("DisplayName"),
```
[the remark this answers is not in the document]
Comment (ParthaI) [reply oci/table_oci_autoscaling_auto_scaling_policy.go:83]: No, We do not have tag support for.

File: oci/table_oci_autoscaling_auto_scaling_policy.go [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+	session, err := autoScalingService(ctx, d, region)
+	if err != nil {
+		logger.Error("oci_autoscaling_auto_scaling_policy.listAutoscalingAutoScalingPolicies", "connection_error", err)
+		return nil, err
+	}
+
+	//Build request parameters
+	request := buildAutoscalingAutoScalingPolicyFilters(equalQuals)
+	request.Limit = types.Int(100)
+	request.AutoScalingConfigurationId = configuration.Id
```
Comment (misraved) [inline oci/table_oci_autoscaling_auto_scaling_policy.go:136]: @ParthaI did we miss adding a check for `auto_scaling_configuration_id` optional qual?

**Review by misraved — COMMENTED** [review]
@ParthaI please take a look at the feedback. Thanks!!!

File: oci/table_oci_autoscaling_auto_scaling_policy.go (hunk 8 above)
Comment (ParthaI) [reply oci/table_oci_autoscaling_auto_scaling_policy.go:142]: Added a check for it.

https://github.com/turbot/steampipe-plugin-oci/pull/513/files#diff-e2964d821977ddd037fda883617dadd4ab9e92f5716f0571881f14f2209a7da7R127

**Review by misraved — APPROVED** [review]

**Pull request merged**